### ISSUE TITLE
About the function of group_filter_format

### ISSUE THREAD

**Issue by takasead** [adobe-apiplatform/user-sync.py#658]
A customer asked me about "group_filter_format" in the "connector-ldap.yml" configuration file.
"Group_filter_format" searches the group information of Open LDAP server and Active Diretctory of Windows.
And, in the "user-sync-config.yml" file, if it is a mapped group, it is a function used to map with the user group of Admin Console.
I recognize it like this. Is my idea correct?

**Comment by Luci2015** [comment]
It is a filter to check if the groups you name in command line or in the group mapping section of the user-sync-config file really exist. It is not involved in the direct mapping to Admin Console groups (that depends on being a member or not of the source mapped groups).

**Comment by takasead** [comment]
@Luci2015


Thank you for answering.

I got it
Sure, I commented out "group_filter_format" in the "connector-ldap.yml" file and ran UST, but I was able to successfully map to the Admin Console user group.
I found that it had nothing to do with the mapping function.

In that case,
Is there any problem in operation even if "group_filter_format" is commented out?
Is there any reason why it would be better to have a "group_filter_format"?

Thank you.

**Comment by Luci2015** [comment]
`group_filter_format` takes a default when you comment it out, this is why it's still working.
`Is there any reason why it would be better to have a "group_filter_format"?`
filters are very important, but that does not mean any situation can be handled with their default values. If you know what you are doing - modify them as you need.

**Comment by takasead** [comment]
@Luci2015

Thank you for answering.

I think you understand "group_filter_format:" in your answer.

I suggest the following to our customers:
-The "group_filter_format:" setting is "a filter that needs to check if the specified group actually exists".
-The setting of "group_filter_format:" is "Use as it is (without changing the setting)"

By the way,
By adding "attribute = attribute value" in "all_users_filter:" of "connector-ldap-fil.yml", only the users who have the attribute value among the users registered in LDAP can be "synchronized target". 
Is it possible to filter by specifying the LDAP attribute value in "group_filter_format:"?

Thank you.

**Comment by Luci2015** [comment]
I do not think you need to do that. Most common scenario is to run the tool with `--users mapped` or have in the invocation_defaults the `users: mapped` statement. This means that ONLY the accounts membrs of the group(s) you specify for each  'directory_group' in user-sync-config file will be part of the sync.
Behind the scene, what happens is a single query formed by a logical AND between the `all_users_filter` and `group_member_filter_format`, which you can 'translate' as "_get the accounts that can be found in provided base_dn and are not status disabled AND which are members of provided LDAP group(s) from user-sync-config_".

If not all the accounts in the mapped LDAP groups are meant to be extracted, you can alter the `all_users_filter` to add an extra condition that eliminates those accounts or make the rest of them to be extracted. 
Alternatively, they can be filtered some more after extraction via the extension config file (again if you can find a pattern to isolate/identify these accounts), and a little bit of coding.

**Comment by takasead** [comment]
@ Luci2015

Thank you for answering.

I'm sorry for the late reply.
Thank you for your polite and detailed explanation.

Last week, I sent an email to the customer based on the information I received.

So far, there are no additional questions from customers about it.
I also tried various things on my experimental server.


From now on, I would like to set filters with "group_member_filter_format" and "all_users_filter:" to control synchronization with the LDAP server.
Thanks to the valuable information, we were able to answer the customer. Thank you very much for helping me.
